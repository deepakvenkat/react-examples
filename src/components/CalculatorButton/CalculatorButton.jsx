import React from 'react';
import Button from '@material-ui/core/Button';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

const styles = {
  root: {
    height: '75px',
    width: '75px',
    borderRadius: 0,
    borderColor: 'red',
    border: 'solid 0.5px',
    fontSize: '28px',
    backgroundColor: (props) => ( props.type === 'number' ? 'white' : 'pink'),
    '&$hover': {
      backgroundColor: (props) => ( props.type === 'number' ? 'white' : 'pink'),
    },
  },
}

const CalculatorButton = ({
  classes,
  text,
}) => (
  <Button
    variant="contained"
    disableElevation
    className={classes.root}
  >
    {text}
  </Button>
);

CalculatorButton.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  // TODO: if type is number, it should also have a value
  type: PropTypes.oneOf(['number', 'operator']).isRequired,
  text: PropTypes.string.isRequired,
};

export default withStyles(styles)(CalculatorButton);
