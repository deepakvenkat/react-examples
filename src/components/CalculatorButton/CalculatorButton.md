### CalculatorButton

#### Number
```jsx
<CalculatorButton
  type="number"
  text="1"
/>
```

#### Operator
```jsx
<CalculatorButton
  type="operator"
  text="+"
/>
```
