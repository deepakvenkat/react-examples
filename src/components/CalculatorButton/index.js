import CalculatorButton from './CalculatorButton.jsx';

export default CalculatorButton;
