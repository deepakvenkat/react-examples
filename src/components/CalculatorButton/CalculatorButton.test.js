import React from 'react';
import { render } from '@testing-library/react';
import CalculatorButton from './CalculatorButton';

test('renders the calculatorbutton with type: number', () => {
  const { asFragment } = render(
    <CalculatorButton
      type="number"
      text="1"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});


test('renders the calculatorbutton with type: operator', () => {
  const { asFragment } = render(
    <CalculatorButton
      type="operator"
      text="1"
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
