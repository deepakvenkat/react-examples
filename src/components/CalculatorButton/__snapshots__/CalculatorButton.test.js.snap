// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the calculatorbutton with type: number 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained CalculatorButton-root-1 CalculatorButton-root-2 MuiButton-disableElevation"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      1
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`renders the calculatorbutton with type: operator 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained CalculatorButton-root-3 CalculatorButton-root-4 MuiButton-disableElevation"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      1
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
